src/main/java/seedu/uninurse/storage/JsonAdaptedTask.java

```diff
@@ -42,6 +52,11 @@ public Task toModelType() throws IllegalValueException {
         if (!Task.isValidTaskDescription(taskDescription) || !DateTime.isValidDateTime(dateTime.getDateTime())) {
             throw new IllegalValueException(Task.MESSAGE_CONSTRAINTS);
         }
-        return new Task(taskDescription, dateTime.toModelType());
+
+        if (freq == null) {
+            return new Task(taskDescription, dateTime.toModelType());
+        } else {
+            return new RecurringTask(taskDescription, dateTime.toModelType(), freq);
+        }
     }
```

Comment:
Same as the suggestion above to use an `Optional<Frequency>#isEmpty()` to avoid checking against null values.